components/Chips/tests/unit/ChipViewRippleTests.m

```diff
@@ -0,0 +1,148 @@
+
+@end
+
+@implementation ChipViewRippleTests
+
+- (void)setUp {
+  [super setUp];
+
+  self.chipView = [[MDCChipView alloc] init];
+}
+
+- (void)tearDown {
+  self.chipView = nil;
+
+  [super tearDown];
+}
+
+/**
+ Test to confirm behavior of initializing a @c MDCChipView without any customization.
+ */
+- (void)testDefaultChipViewBehaviorWithRipple {
+  // Then
+  XCTAssertNotNil(self.chipView.rippleView);
+  XCTAssertEqualObjects([self.chipView.rippleView rippleColorForState:MDCRippleStateNormal],
+                        [UIColor colorWithWhite:0 alpha:(CGFloat)0.12]);
+  XCTAssertEqualObjects([self.chipView.rippleView rippleColorForState:MDCRippleStateHighlighted],
+                        [UIColor colorWithWhite:0 alpha:(CGFloat)0.12]);
+  XCTAssertEqual(self.chipView.rippleView.rippleStyle, MDCRippleStyleBounded);
+  XCTAssertFalse(self.chipView.enableRippleBehavior);
+  XCTAssertNil(self.chipView.rippleView.superview);
+  XCTAssertFalse(self.chipView.rippleView.allowsSelection);
+  CGRect chipViewBounds = CGRectStandardize(self.chipView.bounds);
+  CGRect rippleBounds = CGRectStandardize(self.chipView.rippleView.bounds);
+  XCTAssertTrue(CGRectEqualToRect(chipViewBounds, rippleBounds), @"%@ is not equal to %@",
+                NSStringFromCGRect(chipViewBounds), NSStringFromCGRect(rippleBounds));
+}
+
+/**
+ Test to confirm that setting @c enableRippleBehavior adds the @c rippleView as a subview.
+ */
+- (void)testEnableRippleBehaviorAddsRippleViewAsSubviewOfButton {
+  // When
+  self.chipView.enableRippleBehavior = YES;
+
+  // Then
+  XCTAssertNil(self.chipView.inkView.superview);
+  XCTAssertEqualObjects(self.chipView.rippleView.superview, self.chipView);
+}
+
+/**
+ Test to confirm toggling @c enableRippleBehavior removes the @c rippleView as a subview.
+ */
+- (void)testSetEnableRippleBehaviorToYesThenNoRemovesRippleViewAsSubviewOfButton {
+  // When
+  self.chipView.enableRippleBehavior = YES;
+  self.chipView.enableRippleBehavior = NO;
+
+  // Then
+  XCTAssertEqualObjects(self.chipView.inkView.superview, self.chipView);
+  XCTAssertNil(self.chipView.rippleView.superview);
+}
+
+/**
+ Test setting @c inkColor correctly sets the @c rippleColor on @c rippleView of the chip view.
+ */
+- (void)testSetCustomInkColorUpdatesRippleViewForHighlightedState {
+  // Given
+  UIColor *fakeColor = UIColor.redColor;
+
+  // When
+  [self.chipView setInkColor:fakeColor forState:UIControlStateHighlighted];
+
+  // Then
+  XCTAssertEqualObjects([self.chipView.rippleView rippleColorForState:MDCRippleStateHighlighted],
+                        fakeColor);
+}
+
+/**
+ Test setting @c inkColor to an unsupported stateful ripple state still updates the ripple to the
+ correct color when that state is set.
+ */
+- (void)testNonSupportedRippleStateStillUpdatesRippleColor {
+  // Given
+  UIColor *fakeColor = UIColor.redColor;
+
+  // When
+  [self.chipView setInkColor:fakeColor forState:UIControlStateDisabled];
+  self.chipView.enabled = NO;
```

Comment:
Why doesn't Ripple support disabled states? [Disabled seems to be a Material state](https://material.io/design/interaction/states.html#disabled).  I would expect the Ripple view to handle any necessary overlay/color changes as a consequence of disabling the component.

Reply:
The disabled state does not have any Ripple or Overlay involved in it. However, we probably should allow clients if they wish to add a ripple in the disabled state. This could be added to the Ripple component once we have the state system fully understood by the team.